src/utils/blockedMonitor.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AzureFunctionsRpcMessages as rpc } from './../../azure-functions-language-worker-protobuf/src/rpc';
+import LogCategory = rpc.RpcLog.RpcLogCategory;
+import LogLevel = rpc.RpcLog.Level;
+import blockedAt = require('blocked-at');
+
+export function startBlockedMonitor(channel: { log: (log: rpc.IRpcLog) => void }, threshold = 100): NodeJS.Timer {
+    let blockedHistory: { time: string; duration: number; stack: string; resource: string }[] = [];
+
+    //threshold - minimum miliseconds of blockage to report.
+    //other parameters are default, more details on https://github.com/naugtur/blocked-at.
+    blockedAt(
+        (ms, stack, resource) => {
+            blockedHistory.push({ time: new Date().toUTCString(), duration: ms, stack: stack, resource: resource });
```

Comment:
A threshold of 100 ms feels too small to me. I tested and consistently got false positives lasting around 200-300 ms. I doubt users would notice and file a CRI for a blockage that small anyways. How about 5000 ms?

If we have a larger threshold, we can get rid of the `setInterval` and just log directly from inside `blockedAt`. We also would not need to print the `time` since the time associated with the log would be accurate in that case.